src/context/WalletProvider.tsx

```diff
@@ -16,21 +16,127 @@ import { useBalance } from '../hooks'
 
 declare global {
   interface Window {
-    ethereum: MetaMaskInpageProvider
     solana: any
   }
 }
 
 const WalletProvider = function WalletProvider({ children }: { children: React.ReactNode }) {
   const [state, setState] = useState<IWalletStoreState>(INITIAL_STATE)
 
+  const connectCoinbase = async (chainId: number): Promise<boolean> => {
+    if (!window.ethereum) {
+      return false
+    }
+
+    const ethereum: any  = window.ethereum
+    setState({ ...state, status: WalletStatusEnum.LOADING })
+
+    const findedProvider = () => {
+      if (ethereum.providers?.length) {
+        const provider = ethereum?.providers.find((prov: any) => prov?.isCoinbaseWallet)
+        if (!provider) {
+          throw new Error('[WalletProvider] No provider found')
+        }
+
+        return provider
+      }
+
+      return ethereum
+    }
+
+    try {
+      const CoinbaseWalletSDK = (await import('@coinbase/wallet-sdk')).default
+      const coinbase = new CoinbaseWalletSDK({
+        appName: document.title
+      })
+
+      const rpcUrl = rpcMapping[chainId]
+      const walletProvider = coinbase.makeWeb3Provider(rpcUrl, chainId)
+
+      const provider = new ethers.providers.Web3Provider(walletProvider as unknown as ExternalProvider, 'any')
+      await provider.send('eth_requestAccounts', [])
+
+
+      let { chainId: walletChainId, address, addressShort, addressDomain } = await fetchEvmWalletInfo(provider)
+
+      const isNeedToChangeNetwork = chainId && walletChainId !== chainId
+
+      if (isNeedToChangeNetwork) {
+        const network = getNetworkById(chainId)
+        if (!network.data.params) {
+          setState({ ...state, status: WalletStatusEnum.NOT_INITED })
+          throw new Error(`Missing network ${chainId} params`)
+        }
+        const isChanged = await evmChangeNetwork(network.data.params)
+        if (isChanged) {
+          walletChainId = network.chain_id
+        }
+      }
+
+      walletProvider.on('chainChanged', evmChainChangeHandler as any)
+      walletProvider.on('accountsChanged', evmAccountChangeHandler as any)
+
+      setState(prev => ({
+        ...prev,
+        ...{
+          isConnected: true,
+          status: WalletStatusEnum.READY,
+          name: WALLET_NAMES.Coinbase,
+          provider,
+          walletProvider,
+          chainId: walletChainId,
+          address,
+          addressShort,
+          addressDomain
+        }
+      }))
+
+      localStorage.setItem('web3-wallets-name', WALLET_NAMES.Coinbase)
+      localStorage.setItem(
+        LOCAL_STORAGE_WALLETS_KEY,
+        JSON.stringify({
+          name: WALLET_NAMES.Coinbase,
+          chainId,
+          address: addressDomain || addressShort
+        })
+      )
```

Comment:
Use a separate function for this

Reply:
> Use a separate function for this

I will create an issue with `useLocalStorage`